### DISCUSSION TITLE
Can we use `GITHUB_TOKEN` instead of a custom PAT? If not, why not?

### DISCUSSION THREAD

**Discussion by TomasHubelbauer** [upptime/upptime#780] [Q&A]
I understand that the custom PAT when configured according to the Getting Started guide has some extra scopes which the `GITHUB_TOKEN` PAT might not have but I think it is possible to use the `permissions` keyword in the workflow YAML to extend its scope. I am not certain this is the case but can't you make the built-in PAT as powerful as a custom PAT with this? Or close enough that it would cover all the scopes needed for Upptime to work?

If this was possible it would simplify the Getting Started guide a lot and it would also make it so that actions carried out by Upptime (like the repo updates) wouldn't be associated with the user who made the PAT. In my case that is what happens: https://github.com/TomasHubelbauer (this profile repository is generated using the GitHub API where the endpoint to see all my activity is used and it seems to include entries carried out by Upptime with my PAT).

**Comment by AnandChowdhary** [comment]
I agree, I think we can probably get everything we need done using the `permissions:` key in the workflows themselves. Would be happy to accept a PR!

**Reply by TomasHubelbauer** [reply]
Let's continue in https://github.com/upptime/upptime/pull/781
